substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/code/CodeInfoEncoder.java

```diff
@@ -97,6 +120,53 @@
         final Counter virtualObjectsCount = new Counter(group, "Number of virtual objects", "Number of virtual objects encoded");
     }
 
+    public static final class Encoders {
+        final FrequencyEncoder<JavaConstant> objectConstants;
+        final FrequencyEncoder<Class<?>> sourceClasses;
+        final FrequencyEncoder<String> sourceMethodNames;
+        final FrequencyEncoder<String> names;
+
+        private Encoders() {
+            this.objectConstants = FrequencyEncoder.createEqualityEncoder();
+            this.sourceClasses = FrequencyEncoder.createEqualityEncoder();
+            this.sourceMethodNames = FrequencyEncoder.createEqualityEncoder();
+            sourceMethodNames.addObject("<init>");
```

Comment:
Adding `<init>` once here screws up the frequency of this name, i.e., it will get a high index even though it is encoded frequently. When you added in `prepareMetadataForMethod` for every constructor, then the frequency would be correct.

Reply:
Fixed